### PR TITLE
REST high-level client: parse back _ignored meta field

### PR DESCRIPTION
**Pull request by javanna**
`GetResult` and `SearchHit` have been adjusted to parse back the _ignored meta field whenever it gets printed out. Also expanded the test to make sure this is covered.

Fixed also a small problem around highlighted fields in SearchHitTests.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/search/SearchHitTests.java [hunk 1]
```diff
@@ -19,40 +19,38 @@
...
 package org.elasticsearch.search;
 
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Predicate;
```
Comment (javanna) [inline server/src/test/java/org/elasticsearch/search/SearchHitTests.java]: I can never figure out how to setup my IDE to not have these changes.

**Review by jpountz — APPROVED** [review]
Thanks @javanna !

**Comment by javanna** [comment]
thanks @jpountz !

**Comment by javanna** [comment]
pending backport to 6.4

**Pull request merged**